vertical-pod-autoscaler/e2e/v1/recommender.go

```diff
@@ -226,6 +226,73 @@ var _ = RecommenderE2eDescribe("VPA CRD object", func() {
 	})
 })
 
+var _ = RecommenderE2eDescribe("VPA ContainerScalingMode", func() {
+	f := framework.NewDefaultFramework("vertical-pod-autoscaling")
+
+	var (
+		vpaClientSet *vpa_clientset.Clientset
+	)
+
+	ginkgo.BeforeEach(func() {
+		config, err := framework.LoadConfig()
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		vpaClientSet = vpa_clientset.NewForConfigOrDie(config)
+	})
+
+	ginkgo.It("ContainerScalingModeAuto recommendations", func() {
+		ginkgo.By("Setting up a hamster deployment")
+		createTwoHamestersDeployment(f)
+		vpaCRD := createVpaCRDWithContainerScalingMode(f, vpa_types.ContainerScalingModeAuto)
+
+		ginkgo.By("Waiting for recommendation to be filled")
+		vpa, err := WaitForRecommendationPresent(vpaClientSet, vpaCRD)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(len(vpa.Status.Recommendation.ContainerRecommendations)).Should(gomega.BeNumerically("==", 2))
+
+	})
+
+	ginkgo.It("ContainerScalingModeOff recommendations", func() {
+		ginkgo.By("Setting up a hamster deployment")
+		createTwoHamestersDeployment(f)
+		vpaCRD := createVpaCRDWithContainerScalingMode(f, vpa_types.ContainerScalingModeOff)
+
+		ginkgo.By("Waiting for recommendation to be filled")
+		vpa, err := WaitForRecommendationPresent(vpaClientSet, vpaCRD)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		// hamster container has recommendations turned off,
+		// so we expect only one container recommendation for hamster2.
+		gomega.Expect(len(vpa.Status.Recommendation.ContainerRecommendations)).Should(gomega.BeNumerically("==", 1))
+		gomega.Expect(vpa.Status.Recommendation.ContainerRecommendations[0].ContainerName).To(gomega.Equal("hamster2"))
+	})
+})
+
+func createTwoHamestersDeployment(f *framework.Framework) {
+	d := NewHamsterDeploymentWithResourcesAndLimits(f,
```

Comment:
Since the request and limit doesn't matter much, maybe just use NewHamsterDeployment

Reply:
Done.